### ISSUE TITLE
Obsolete references to URI.escape and URI.unescape

### ISSUE THREAD

**Issue by jacobjlevine** [Empact/roxml#71]
I'm seeing the following two deprecation warnings (Ruby 2.7.1, Rails 6.0.3.1), both from /app/vendor/bundle/ruby/2.7.0/gems/roxml-4.1.0/lib/roxml/xml/references.rb:50:

warning: URI.**unescape** is obsolete
warning: URI.**escape** is obsolete

`URI.unescape(convention.call(URI.escape(what, /\/|::/)))`

**Comment by jacobjlevine** [comment]
@Empact any chance we can get this merged in?

**Comment by ibmandeep** [comment]
@Empact Please merge this PR this should solve the issue.

**Comment by ibrahim-saifee** [comment]
This PR might solve the issue
https://github.com/Empact/roxml/pull/77

**Comment by ibrahim-saifee** [comment]
> I'm seeing the following two deprecation warnings (Ruby 2.7.1, Rails 6.0.3.1), both from /app/vendor/bundle/ruby/2.7.0/gems/roxml-4.1.0/lib/roxml/xml/references.rb:50:
> 
> warning: URI.**unescape** is obsolete
> warning: URI.**escape** is obsolete
> 
> `URI.unescape(convention.call(URI.escape(what, /\/|::/)))`

https://github.com/Empact/roxml/pull/77

**Comment by ibrahim-saifee** [comment]
This PR #77 might resolve the issue. As the `URI.escape` and `URI.unescape` is deprecated for Ruby 2.7.x so, we need to use `URI.encode_www_form_component` and `URI.decode_www_form_component` for URL encoding and decoding respectively.

https://ruby-doc.org/stdlib-2.5.3/libdoc/uri/rdoc/URI/Escape.html#method-i-escape
https://ruby-doc.org/stdlib-2.5.3/libdoc/uri/rdoc/URI/Escape.html#method-i-unescape

**Comment by mmplisskin** [comment]
Any possibility for a review on this @Empact

**Comment by nehresma** [comment]
@Empact first, thank you very much for this gem.  It has been very helpful to a number of us in the community over the years.

There are several somewhat large and often used downstream projects (like quickbooks-ruby) using this gem that are impacted by this particular issue.  Those projects are unable to upgrade to Ruby 3 because `URI.escape` and `URI.unescape` have been removed.  Would you have a few minutes to merge this in and release an updated gem?  Or would you be open to considering adding someone else in the community as a collaborator on this github repository and the rubygems.org gem so that an update could be released?

Thanks again for this gem and for your consideration.

**Comment by anthonyhamou** [comment]
@Empact first, thank you very much for this gem. It has been very helpful to a number of us in the community over the years.

Could you please help us ? it is urgent !!

**Comment by jacobjlevine** [comment]
Hey folks! I just got in touch with @Empact, and he's open to giving an additional contributor admin permissions for the gem. Anyone open to taking on that responsibility?

**Comment by campbecf** [comment]
I'd be willing to contribute to the project. Some of the issues on this repo are blockers for my company. How do I go about getting in touch with @Empact  ?

**Comment by lostapathy** [comment]
I have commit to this repo but have been hesitant to merge much since we can't cut a release.  If somebody gets the ability to push new releases to rubygems, I'd be happy to help more with the maintenance here.

**Comment by Empact** [comment]
Hello everyone, sorry about the neglect of this library but I'm not currently using it so it's low on my list of priorities. @lostapathy can merge, and I've added @campbecf as well. I'm happy to extend the ability to cut releases as well.

**Comment by Empact** [comment]
Closed via #77

**Comment by jacobjlevine** [comment]
Hey @Empact @lostapathy @campbecf! Looks like this has been merged into master but hasn't been cut as a release yet. Would it be possible to make that happen to unblock those of us looking to explore upgrading to Ruby 3? Thanks!

**Comment by lostapathy** [comment]
Unfortunately I don't have the ability to cut a new gem - I think only @Empact has that.

**Comment by anthonyhamou** [comment]
Hello Ben @Empact we need you to cut roxml as a release please. Can we rely on you ? It is very important ! Thx

**Comment by Empact** [comment]
Thanks everyone! Please see 4.2.0, and let me know if you run into any issues:
https://github.com/Empact/roxml/releases/tag/v4.2.0
https://rubygems.org/gems/roxml

**Comment by pmsiam0** [comment]
> This PR #77 might resolve the issue. As the `URI.escape` and `URI.unescape` is deprecated for Ruby 2.7.x so, we need to use `URI.encode_www_form_component` and `URI.decode_www_form_component` for URL encoding and decoding respectively.
> 
> https://ruby-doc.org/stdlib-2.5.3/libdoc/uri/rdoc/URI/Escape.html#method-i-escape https://ruby-doc.org/stdlib-2.5.3/libdoc/uri/rdoc/URI/Escape.html#method-i-unescape

This one isn't worked in my case cause I use ruby 3.0.4p208 and my case it is not working can anyone help me to do this in my code?
